clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/DaVinciRecordTransformer.java

```diff
@@ -34,6 +34,18 @@ public interface DaVinciRecordTransformer<K, V, O extends TransformedRecord> {
    */
   O put(Lazy<K> key, Lazy<V> value);
 
+  /**
```

Comment:
I have some questions about these two functions:
```
  Schema getKeyOutputSchema();
  Schema getValueOutputSchema();
```
Will these schemas be the ones used to persist the transformed data in the local storage?
If yes, that might introduce some risks.
Today, DVC will leverage the on-disk snapshot during restart/deployment and if the previous snapshot is using a different schemas because of logic change after a new deployment, the newly deployed DVC app shouldn't leverage the previous snapshot, and even with the same schemas, the transforming logic can be changed, and keeping ingesting on top of the previous snapshot can be problematic as well.

If we want to go with this direction, I think we should version the transformer, which should cover both schemas and transformation logic.
At the end of the day, it will be still a best effort.

A naive way to address this concern is to bootstrap every time, which might not be ideal for some use case, which want to leverage the previous snapshot to speed up bootstrapping.

Reply:
Right now we have an additional check on restart whether the schema has changed compared to the on-disk snapshot or if a transformer is present in the new deployment when it previously wasn't and will reingest. The reingestion part is what is meant by bootstrapping correct? Although this doesn't account for change in transformer logic.

Reply:
I can add a version to the transformer, that we can state the user should pass a monotonically increasing version number when they update with a new transformer. On restart we can just check this instead.

Reply:
Maybe we can leverage the snapshot if there was no transformation before, and apply the new transformation logic locally instead of ingestion over a network?

Reply:
Let's take this up in another PR.  I think the idea of a versioning scheme outside of a schema change makes sense, but let's think on how best to approach that.